core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -0,0 +1,344 @@
+    @Nullable
+    private DnsQueryLifecycleObserverFactory dnsQueryLifecycleObserverFactory;
+    @Nullable
+    private List<String> searchDomains;
+    @Nullable
+    private Integer ndots;
+    @Nullable
+    private Boolean decodeIdn;
+
+    DnsResolverGroupBuilder() {}
+
+    /**
+     * Sets {@link Backoff} which is used when the {@link DnsNameResolver} fails to update the cache.
+     */
+    public DnsResolverGroupBuilder refreshBackoff(Backoff refreshBackoff) {
+        this.refreshBackoff = requireNonNull(refreshBackoff, "refreshBackoff");
+        return this;
+    }
+
+    /**
+     * Sets the timeout of refreshing DNS caches. After this timeout, the previously resolved
+     * DNS cache is invalidated.
+     */
+    public DnsResolverGroupBuilder refreshTimeout(Duration refreshTimeout) {
+        requireNonNull(refreshTimeout, "refreshTimeout");
+        checkArgument(!refreshTimeout.isNegative(), "refreshTimeout: %s (expected: >= 0)", refreshTimeout);
+        return refreshTimeoutMillis(refreshTimeout.toMillis());
+    }
+
+    /**
+     * Sets the timeout of refreshing DNS caches in milliseconds. After this timeout,
+     * the previously resolved DNS cache is invalidated.
+     */
+    public DnsResolverGroupBuilder refreshTimeoutMillis(long refreshTimeoutMillis) {
+        checkArgument(refreshTimeoutMillis >= 0, "refreshTimeoutMillis: %s (expected: >= 0)",
+                      refreshTimeoutMillis);
+        this.refreshTimeoutMillis = refreshTimeoutMillis;
+        return this;
+    }
+
+    /**
+     * Sets the minimum and maximum TTL of the cached DNS resource records in seconds. If the TTL of the DNS
+     * resource record returned by the DNS server is less than the minimum TTL or greater than the maximum TTL,
+     * this resolver will ignore the TTL from the DNS server and use the minimum TTL or the maximum TTL instead
+     * respectively.
+     * The default value is {@code 1} and {@link Integer#MAX_VALUE}, which practically tells this resolver to
+     * respect the TTL from the DNS server.
+     */
+    public DnsResolverGroupBuilder ttl(int minTtl, int maxTtl) {
+        checkArgument(minTtl > 0 && minTtl <= maxTtl,
+                      "minTtl: %s, maxTtl: %s (expected: 1 <= minTtl <= maxTtl)", minTtl, maxTtl);
+        this.minTtl = minTtl;
+        this.maxTtl = maxTtl;
+        return this;
+    }
+
+    /**
+     * Sets if this resolver should generate the detailed trace information in an exception message so that
+     * it is easier to understand the cause of resolution failure. This flag is enabled by default.
+     */
+    public DnsResolverGroupBuilder traceEnabled(boolean traceEnabled) {
+        this.traceEnabled = traceEnabled;
+        return this;
+    }
+
+    /**
+     * Sets the cache for {@code CNAME} mappings.
+     */
+    public DnsResolverGroupBuilder cnameCache(DnsCnameCache cnameCache) {
+        this.cnameCache = requireNonNull(cnameCache, "cnameCache");
+        return this;
+    }
```

Comment:
Question: Do we need to expose this property? Could you explain why we allow a user to specify this while disabling other caches?

Reply:
Removed. 😃